[Bug]: sabnzbd 4.3.1-1 fails to install due to sabctools / missing Android NDK

### Problem description

The PR: https://github.com/termux/termux-packages/pull/20096/files introduced the installation of sabctools (which is the same as python-sabyenc3 as far as I understand it) through pip which fails to due the missing `cpu-features.h` from the Android NDK. I believe this was also the reason why python-sabyenc3 was introduced as a separate package when sabnzbd was introduced in https://github.com/termux/termux-packages/pull/16129

Terminal output during update/installation:

```
Preparing to unpack .../sabnzbd_4.3.1-1_all.deb ...
Unpacking sabnzbd (4.3.1-1) over (4.2.3) ...
Setting up sabnzbd (4.3.1-1) ...
Installing dependencies through pip...
Collecting apprise==1.7.6
  Downloading apprise-1.7.6-py3-none-any.whl.metadata (44 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.8/44.8 kB 2.6 MB/s eta 0:00:00
Collecting sabctools==8.1.0
  Downloading sabctools-8.1.0.tar.gz (234 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 234.7/234.7 kB 4.7 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
[...]
Downloading urllib3-2.2.1-py3-none-any.whl (121 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 121.1/121.1 kB 10.2 MB/s eta 0:00:00
Building wheels for collected packages: sabctools, CT3, cffi, PyYAML
  Building wheel for sabctools (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for sabctools (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [41 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-aarch64-cpython-311
      creating build/lib.linux-aarch64-cpython-311/sabctools
      copying src/__init__.py -> build/lib.linux-aarch64-cpython-311/sabctools
      copying src/sabctools.pyi -> build/lib.linux-aarch64-cpython-311/sabctools
      copying src/py.typed -> build/lib.linux-aarch64-cpython-311/sabctools
      running build_ext
      ==> Baseline detection: ARM=True, x86=False, macOS=False
      ==> Checking support for flag: -std=c++11
      ==> Please ignore any errors shown below!
      creating data
      creating data/data
      creating data/data/com.termux
      creating data/data/com.termux/files
      creating data/data/com.termux/files/usr
      creating data/data/com.termux/files/usr/tmp
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.11 -c /data/data/com.termux/files/usr/tmp/tmpifnb35mx.cc -o data/data/com.termux/files/usr/tmp/tmpifnb35mx.o -std=c++11
      ==> Success!
      ==> Checking support for include: sys/auxv.h
      ==> Please ignore any errors shown below!
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.11 -c /data/data/com.termux/files/usr/tmp/tmpcyclf693.cc -o data/data/com.termux/files/usr/tmp/tmpcyclf693.o
      ==> Success!
      ==> Checking support for define: __aarch64__
      ==> Please ignore any errors shown below!
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.11 -c /data/data/com.termux/files/usr/tmp/tmp9vy2qa3q.cc -o data/data/com.termux/files/usr/tmp/tmp9vy2qa3q.o
      ==> Success!
      ==> Checking support for flag: -march=armv8-a+crc
      ==> Please ignore any errors shown below!
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.11 -c /data/data/com.termux/files/usr/tmp/tmpuqni31td.cc -o data/data/com.termux/files/usr/tmp/tmpuqni31td.o -march=armv8-a+crc
      ==> Success!
      creating build/src
      creating build/src/yencode
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -Isrc/crcutil-1.0/code -Isrc/crcutil-1.0/examples -I/data/data/com.termux/files/usr/include/python3.11 -c src/yencode/platform.cc -o build/src/yencode/platform.o -Wall -Wextra -Wno-unused-function -fomit-frame-pointer -fno-rtti -fno-exceptions -O3 -fPIC -fwrapv -std=c++11
      src/yencode/platform.cc:5:12: fatal error: 'cpu-features.h' file not found
          5 | #  include <cpu-features.h>
            |            ^~~~~~~~~~~~~~~~
      1 error generated.
      error: command '/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for sabctools
  Building wheel for CT3 (setup.py) ... done
  Created wheel for CT3: filename=CT3-3.3.3.post1-py3-none-any.whl size=176348 sha256=a9be88e14efe7c62d1beeccaa047130572719b68ecf478359cfa24e2b70c3b89
  Stored in directory: /data/data/com.termux/files/home/.cache/pip/wheels/12/e6/23/21fe671ee6065b6b69d07c5f948b39a55605378e76de760b67
  Building wheel for cffi (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for cffi (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [38 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-aarch64-cpython-311
      creating build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/pkgconfig.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/cparser.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/setuptools_ext.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/backend_ctypes.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/_imp_emulation.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/model.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/vengine_cpy.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/cffi_opcode.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/_shimmed_dist_utils.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/ffiplatform.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/lock.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/error.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/verifier.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/__init__.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/commontypes.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/recompiler.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/api.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/vengine_gen.py -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/_cffi_include.h -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/parse_c_type.h -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/_embedding.h -> build/lib.linux-aarch64-cpython-311/cffi
      copying src/cffi/_cffi_errors.h -> build/lib.linux-aarch64-cpython-311/cffi
      running build_ext
      building '_cffi_backend' extension
      creating build/temp.linux-aarch64-cpython-311
      creating build/temp.linux-aarch64-cpython-311/src
      creating build/temp.linux-aarch64-cpython-311/src/c
      aarch64-linux-android-clang -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -DFFI_BUILDING=1 -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/data/data/com.termux/files/usr/include -I/data/data/com.termux/files/usr/include/python3.11 -c src/c/_cffi_backend.c -o build/temp.linux-aarch64-cpython-311/src/c/_cffi_backend.o
      aarch64-linux-android-clang -shared -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -fopenmp -static-openmp -fno-openmp-implicit-rpath -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now -Wl,--no-as-needed,-landroid-support,--as-needed -L/home/builder/.termux-build/_cache/android-r26b-api-24-v3/sysroot/usr/lib -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -fopenmp -static-openmp -fno-openmp-implicit-rpath -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now -Wl,--no-as-needed,-landroid-support,--as-needed -L/home/builder/.termux-build/_cache/android-r26b-api-24-v3/sysroot/usr/lib build/temp.linux-aarch64-cpython-311/src/c/_cffi_backend.o -L/data/data/com.termux/files/usr/lib -L/data/data/com.termux/files/usr/lib -lffi -lpython3.11 -o build/lib.linux-aarch64-cpython-311/_cffi_backend.cpython-311.so
      aarch64-linux-android-clang: error: unknown argument: '-fno-openmp-implicit-rpath'
      aarch64-linux-android-clang: error: unknown argument: '-fno-openmp-implicit-rpath'
      error: command '/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cffi
 [...] 
Failed to build sabctools cffi
ERROR: Could not build wheels for sabctools, cffi, which is required to install pyproject.toml-based projects
dpkg: error processing package sabnzbd (--configure):
 installed sabnzbd package post-installation script subprocess returned error exit status 1
Errors were encountered while processing:
 sabnzbd
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)
```


Sadly I'm not familiar with the build system for termux apps to debug/test it myself but if I had to take a guess it could be enough to change

https://github.com/termux/termux-packages/pull/20096/files#diff-8bfd907d1e883b0686edf68e8148d598c901060980e67b631428715b378b8ca2R25
```
if [ -z "$dep" ]; then
```
to
```
if ( [ -z "$dep" ] || [ -z "${dep/sabctools*}" ]);
```
or something similar as that should skip `sabctools` which are already provided by `python-sabyenc3`? 


### What steps will reproduce the bug?

Try to update sabnzbd <4.3.0 to 4.3.1-1 or probably just try to install it. 

### What is the expected behavior?

sabnzbd should be installed

### System information

```shell
Termux Variables:
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=18386
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.mentality.rip/termux-main stable main
# root-repo (sources.list.d/root.list)
deb https://termux.mentality.rip/termux-root root stable
Updatable packages:
All packages up to date
termux-tools version:
1.42.1
Android version:
14
Kernel build information:
Linux localhost 4.9.337-perf-g56f4118d056c #1 SMP PREEMPT Fri May 3 21:35:49 UTC 2024 aarch64 Android
Device manufacturer:
Xiaomi
Device model:
POCO F1
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
```

--- a/packages/sabnzbd/build.sh
+++ b/packages/sabnzbd/build.sh
@@ -4,7 +4,7 @@ TERMUX_PKG_LICENSE="GPL-2.0, GPL-3.0"
 TERMUX_PKG_LICENSE_FILE="LICENSE.txt, GPL2.txt, GPL3.txt"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="4.3.1"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://github.com/sabnzbd/sabnzbd/releases/download/${TERMUX_PKG_VERSION}/SABnzbd-${TERMUX_PKG_VERSION}-src.tar.gz
 TERMUX_PKG_SHA256=c9902c212df3e6b7208c850e6ceab244afc4b3e173459c425db9be4df902bd44
 TERMUX_PKG_AUTO_UPDATE=true
@@ -22,7 +22,8 @@ termux_step_post_get_source() {
 			break
 		fi
 		dep="${dep/[# ]*}"
-		if [ -z "$dep" ]; then
+		# https://github.com/termux/termux-packages/issues/20229
+		if [ -z "$dep" ] || [ -z "${dep/sabctools*}" ]; then
 			continue
 		fi
 		TERMUX_PKG_PYTHON_TARGET_DEPS+="'$dep', "

